### PR TITLE
jenish/feature/athena iris i2c programming

### PR DESCRIPTION
**Pull request by JenishPatel99**
Feature includes:
- I2C protocol for programming iris. Adapted from https://www.st.com/resource/en/application_note/an4221-i2c-protocol-used-in-the-stm32-bootloader-stmicroelectronics.pdf

Scope of PR
- Base code for iris i2c programming code. Error handling was not part of this PR

Note:
- Code was verified on working iris without i2c buffer, but still problems exist with i2c buffer. There has been changes to the code since the last test, and this code is not verified on a iris yet

### CODE REVIEW

File: ex2_hal/iris/equipment_handler/include/iris_i2c.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *  Created on: May 10, 2022
+ *      Author: jenish
+ */
+
+#ifndef INCLUDE_IRIS_I2C_H_
+#define INCLUDE_IRIS_I2C_H_
+
+#include "FreeRTOS.h"
+
+#define SLAVE_ADDR 0x42
```
Comment (Taylorr82) [inline ex2_hal/iris/equipment_handler/include/iris_i2c.h:26]: Should make this name based on IRIS to make it more obvious

File: main/system.h [hunk 2]
```diff
@@ -89,6 +89,7 @@
 #else // These values are expected to be adjusted based on what the developer is working on
 #define IRIS_CONFIG_SPI spiREG4 // SPI1
 #define IRIS_SPI spiREG5        // SPI3
+#define IRIS_I2C i2cREG1        // I2C1
```
Comment (Taylorr82) [inline main/system.h:92]: This should be in an I2C section of system.h, not the SPI section

File: ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+ */
+
+#include "iris_bootloader_cmds.h"
+#include "i2c_io.h"
+#include "HL_gio.h"
+#include "HL_reg_het.h"
+#include "iris_i2c.h"
+#include "FreeRTOS.h"
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (Taylorr82) [inline ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c:28]: Unused include

File: ex2_hal/iris/equipment_handler/source/iris_i2c.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ *  Created on: May 10, 2022
+ *      Author: jenish
+ */
+
+#include "iris_i2c.h"
+#include "FreeRTOS.h"
+#include "i2c_io.h"
+#include "system.h"
+#include <stdio.h>
```
Comment (Taylorr82) [inline ex2_hal/iris/equipment_handler/source/iris_i2c.c:25]: Unused include

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

**Comment by Taylorr82** [comment]
Some sys_logs around here would probably be good

File: ex2_hal/iris/equipment_handler/source/iris_i2c.c (hunk 4 above)
Comment (JenishPatel99) [reply ex2_hal/iris/equipment_handler/source/iris_i2c.c:25]: Sorry, which include is unused

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**